shell/server/pre-meteor.js

```diff
@@ -210,7 +210,24 @@ function checkMagic(buf, magic) {
   return true;
 }
 
-function serveStaticAsset(req, res) {
+function serveSelfTest(req, res) {
+  inMeteor(() => {
+    if (req.method === 'GET' &&
+        req.url === '/') {
+      return serveStaticAsset(req, res, {
```

Comment:
Note that you don't actually care about getting the same etag behavior as the static asset domain, since these are single-use host/filename pairs.  You only have one content type, you don't want to accept POSTs, and since you don't allow arbitrary user content and serve up text-plain, you don't need to worry about the rest of the CSP headers.

Maybe you need the Access-Control-Allow-Origin header, and the Content-Length header too.